relion/protocols/protocol_refine3d.py

```diff
@@ -179,8 +180,11 @@ def _fillDataFromIter(self, imgSet, iteration):
     def _updateParticle(self, particle, row):
         self.reader.setParticleTransform(particle, row)
 
-        if not hasattr(particle, '_rlnRandomSubset'):
-            particle._rlnRandomSubset = Integer()
-
-        particle._rlnRandomSubset.set(row.rlnRandomSubset)
-
+        if self.reader._first:
```

Comment:
This should be handled in the reader...seems like duplicated code...and make use of the internal _first property. I can help with this